cupy/core/_routines_math.pyx

```diff
@@ -95,12 +102,17 @@ cdef ndarray _ndarray_prod(ndarray self, axis, dtype, out, keepdims):
 
 cdef ndarray _ndarray_sum(ndarray self, axis, dtype, out, keepdims):
     for accelerator in _accelerator._routine_accelerators:
+        result = None
         if accelerator == _accelerator.ACCELERATOR_CUB:
             # result will be None if the reduction is not compatible with CUB
             result = cub.cub_reduction(
                 self, cub.CUPY_CUB_SUM, axis, dtype, out, keepdims)
-            if result is not None:
-                return result
+        if accelerator == _accelerator.ACCELERATOR_CUTENSOR:
```

Comment:
I think you need to check if `cutensor` is None?

Reply:
does _accelerator allows to set `ACCELERATOR_CUTENSOR` if cutensor is not available?
If such case then just checking this is fine.

Reply:
I don't want the routines to fallback to CuPy's default reduction silently in such cases. I will fix `_set_routine_accelerator` in another PR.

Reply:
By "in such cases" did you mean the library is absent but the user still requests to use it? If so, I think the current implementation makes sense!